package/MDAnalysis/analysis/helix_analysis.py

```diff
@@ -0,0 +1,515 @@
+    screw_angles, alt_angles = np.arccos(np.clip(cos, -1, 1))  # (2, n_vec)
+
+    # the effect of the below is to move angles into the (pi/4, 3pi/4 domain)
+    quarter_pi, half_pi = np.pi/4, np.pi/2
+    # replace low angles
+    low = screw_angles < quarter_pi
+    screw_angles[low] = np.where(alt_angles[low] < half_pi,
+                                 half_pi - alt_angles[low],
+                                 alt_angles[low] - half_pi)
+    # replace high angles
+    high = screw_angles > 3*quarter_pi
+    screw_angles[high] = np.where(alt_angles[high] < half_pi,
+                                  half_pi + alt_angles[high],
+                                  3*half_pi - alt_angles[high])
+
+    ortho = np.cross(ref_1, rotation_vectors)
+    cos_theta = np.matmul(best_fit, ortho.T)/(pnorm(ortho)*ref_norms[0])
+    angle_to_best_fit = np.arccos(np.clip(cos_theta, -1, 1))
+    screw_angles[angle_to_best_fit < half_pi] *= -1
+    return np.rad2deg(screw_angles)  # (n_vec,)
+
+
+def helix_analysis(positions, ref_axis=[0, 0, 1]):
+    r"""
+    Calculate helix properties from atomic coordinates.
+
+    Each property is calculated from a sliding window of 4 atoms, 
+    from i to i+3.
+
+    Parameters
+    ----------
+    positions: :class:`numpy.ndarray` of shape (N, 3)
+        Atomic coordinates.
+    ref_axis: array-like of length 3, optional
+        The reference axis used to calculate the tilt of the vector of best fit, 
+        and the local screw angles.
+
+    Returns
+    -------
+    dict with the following keys:
+        local_twists: array, shape (N-3,)
+            local twist angle from atom i+1 to i+2
+        residues_per_turn: array, shape (N-3,)
+            number of residues per turn, based on local_twist
+        local_axes:  array, shape (N-3, 3)
+            the length-wise helix axis of the local window
+        local_bends: array, shape (N-6,)
+            the angles between local helix angles, 3 windows apart
+        local_heights: array, shape (N-3,)
+            the rise of each local helix
+        local_rotation_vectors: array, shape (N-2, 3)
+            the unit vector from each local origin to atom i+1
+        local_origins: array, shape (N-2, 3)
+            the projected origin for each helix
+        all_bends: array, shape (N-3, N-3)
+            angles between each local axis
+        global_axes: array, shape (3,)
+            vector of best fit through origins
+        local_screw: array, shape (N-2,)
+            angles of each rotation vector to normal plane of global axes
+    """
+
+    #          ^               ^
+    #           \             / bi
+    #            \           /
+    #         CA_i+2 <----- CA_i+1
+    #         /    \       /   ^
+    #        /    r \     /     \
+    #     V /        \ θ /       \
+    #      /          \ /       CA_i
+    #     v           origin
+    #   CA_i+3
+    #
+    # V: vectors
+    # bi: bisectors
```

Comment:
This might be a bit nit-picky - given the way they're defined below the 'biscectors' are only true biscectors if the CA_i -> CA_i+1 etc vectors have the same length in this plane. This isn't a bad assumption and as far as I can tell the radius calculation later on assumes this anyway, and I don't know what a better name would be; but might be worth clarifying here/in the notebook that they're not directly calculated as biscectors (assuming I'm doing my geometry right of course - do ignore me if I'm wrong!)

Reply:
No, you're right -- they're not normalised here because otherwise the cylinder radius winds up being dependent only on the angle between the inaccurately named ``bisectors``, instead of including the length as well, which made sense to me. It isn't a variable that users can see without looking at the code anyway -- I could rename it to something like ``dvectors`` or ``in_between``, or even just add a comment clarifying that they're not real bisectors?

Reply:
Yeah, fair - I think bisectors probably does actually work better as a descriptive name to get across what they're being used for, so that's fine.